packages/core/src/storage/vectorStore/ChromaVectorStore.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  AddParams,
+  ChromaClient,
+  ChromaClientParams,
+  Collection,
+  Embeddings,
+  IncludeEnum,
+  QueryResponse,
+  Where,
+  WhereDocument,
+} from "chromadb";
+import { BaseNode, Document, MetadataMode } from "../../Node";
+import {
+  VectorStore,
+  VectorStoreQuery,
+  VectorStoreQueryMode,
+  VectorStoreQueryResult,
+} from "./types";
+import { nodeToMetadata } from "./utils";
+
+type ChromaDeleteOptions = {
+  where?: Where;
+  whereDocument?: WhereDocument;
+};
+
+type ChromaQueryOptions = {
+  whereDocument?: WhereDocument;
+};
+
+export class ChromaVectorStore implements VectorStore {
+  DEFAULT_TEXT_KEY = "text";
+  storesText: boolean = true;
+  flatMetadata: boolean = true;
+  private chromaClient: ChromaClient;
+  private collection: Collection | null = null;
+  private collectionName: string;
+
+  constructor(collectionName: string, chromaClientParams?: ChromaClientParams) {
+    this.clearCollection();
+    this.collectionName = collectionName;
+    this.chromaClient = new ChromaClient(chromaClientParams);
+  }
+
+  client(): ChromaClient {
+    return this.chromaClient;
+  }
+
+  // Singleton pattern to ensure we only create one collection
```

Comment:
remove, this is not a singleton pattern, it's just lazily getting the collection